caddyhttp/httpserver/pathcleaner.go

```diff
@@ -0,0 +1,75 @@
+package httpserver
+
+import (
+	"math/rand"
+	"path"
+	"strings"
+	"time"
+)
+
+// A proxy function to prevent one or more of the path cleanup operations:
+//   - collapse multiple slashes into one
+//   - eliminate "/." (current directory)
+//   - eliminate "<parent_directory>/.."
+// by masking certain patterns in the path with a temporary random string.
+// This could be helpful when certain patterns in the path are desired
+// that would otherwise be changed in the path clean up process.
+// One such use case is the presence of the double slashes as protocol separator
+// (e.g., /api/endpoint/http://example.com).
+// This is a common pattern in many applications to allow passing URIs as path argument
+func CleanMaskedPath(p string, mask ...string) string {
```

Comment:
Can you choose more descriptive variable names? Rename `p` to `reqPath`, for example, and `mask` to `masks`...